### DISCUSSION TITLE
Messaging add-on tree view

### DISCUSSION THREAD

**Discussion by RokCoder** [ScratchAddons/ScratchAddons#1562] [Ideas]
It would be great if the messaging add-on featured a collapsible tree view. As it is I quite often miss replies to comments I've left on individual profiles. I get a lot of noise from certain projects that I generally don't read (as it gets tedious posting "Have you tried reading the in-game instructions?", "No, the bots don't cheat" and "It's because you're a New Scratcher") which makes it very easy to miss responses on various profiles.

**Comment by WorldLanguages** [comment]
As far as I know, I made it so profile comments always go first? Then studio comments, then project comments. So replies on other profiles should always be below "your profile" comments, and below studio/project comments.

**Comment by RokCoder** [comment]
Actually, you're right. I'm probably thinking of when I'm waiting for replies when I'm helping people on their projects. Either way, I'd find a collapsible view very helpful. I'm having to scroll through a lot of messages to skip my profile and various projects (and often miss replies on other projects because of this). It might not be so useful for people getting less notifications on their projects but I don't see that a collapsible view would be a bad prospect even if that's the case. I don't know how easy it would be to implement though...

**Reply by WorldLanguages** [reply]
Well, that's a good point. Right now, projects are ordered this way: projects with new comments, then projects with no comments. Maybe they should be ordered like this: projects not by you with new comments, projects by you with new comments, projects with no comments (that need to be yours obviously bc you can't get notified for loves/faves in someone else's project).
I know you're suggesting to make projects, studios and profiles collapsible but it's still a good idea to maintain good ordering.

**Comment by cobaltt7** [comment]
Why not add a setting to let the user configure the order?
________________________________

**Reply by WorldLanguages** [reply]
Easier said than done, what would the settings actually be? IMO this is one of the things that shouldn't be chosen by the user. If they care so deeply about this ordering they are free to fork the addon.
